corehq/apps/es/README.rst

```diff
@@ -94,21 +94,22 @@ Design Details
 Reindex Procedure Details
 '''''''''''''''''''''''''
 
-1. Configure multiplexing on an index.
-
-   - Configure the document adapter for the index with a "secondary index name".
-     This will cause the adapter to use multiplexing logic instead of a single
-     index.
-
-     **Note**: The multiplexing logic required for this operation is not yet
-     implemented. The multiplexing adapter will most likely delegate to two
-     document adapters configured for separate indexes. Suffice it to say that
-     when a secondary index is defined for an adapter, it effectively becomes a
-     multiplexing adapter (which to the consumer, is indistinguishable from a
-     "standard" adapter).
+1. Configure multiplexing on an index by passing in `secondary` index name to `create_document_adapter`.
 
+   - Ensure that there is a migration in place for creating the index.
+   **Note** Tooling around this is a WIP
    - *(Optional)* If the reindex involves other meta-index changes (shards,
      mappings, etc), also update those configurations at this time.
+    **Note** Currently the Adapter will not support reindexing on specific environments but it would be compatible to accommodate it in future. This support will be added once we get to V5 of ES.
```

Comment:
If the index names (for primary and secondary) are set via settings rather than hardcoded, you may be able to add this pretty easily. The secondary can be optional and default to `None`, which is basically how this already works. Then the presence of a secondary index in settings would determine whether or not it multiplexed, and that could force us to make all indices configurable in settings, which we should probably be doing anyway.

Reply:
Yep. You are correct. The code can be easily updated to support setting index names via settings, I guess we would be taking it as future work where we would specifically focus on that. For the time being we would be going ahead with it.

Reply:
@AmitPhulera @joxl can you confirm that the current plan will support upgrading 3rd party environments independently from our own.

This comment suggests that it won't be supported until V5 but that doesn't seem correct.

Reply:
Right now it is not supported and it is easy to support.

Since we don't have the same mappings in all the environments. So we have to reindex at least once in all the environments to ensure that we have the same mappings everywhere.
@joel can correct me if I am wrong but I believe that once that is done then we will add that support.

Reply:
> So we have to reindex at least once in all the environments 

Does this not include 3rd party envs?

Reply:
> **AmitPhulera**
> I believe that once that is done then we will add that support

This is something that _can_ be added after the v5 upgrade, but will not be added as part of the upgrade work.

> **snopoke**
> Does this not include 3rd party envs?

Yes. All envs will be required to reindex.

Reply:
The reindex is required to get all existing indexes to be "created by" the current Elasticsearch version. This is a prerequisite to an Elasticsearch server major version upgrade.